cmake/variables.cmake

```diff
@@ -153,13 +153,33 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
   list(APPEND ML_COMPILE_DEFINITIONS BOOST_ALL_NO_LIB)
 endif()
 
-if(UNIX AND "$ENV{ML_DEBUG}" AND "$ENV{ML_COVERAGE}")
+# Perform a "Release" build by default...
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Release)
+endif()
+
+# However, to keep in step with our historical
+# build system if ML_DEBUG is set to a boolean
+# true value, e.g. ML_DEBUG=1, then override
+# the build type to be Debug
+if("$ENV{ML_DEBUG}")
+  set(CMAKE_BUILD_TYPE Debug)
```

Comment:
I have a slight worry that this is going to cause CMake to want to use the `/MDd` option on Windows instead of `/MD`.  We always want to use `/MD`.

It seems like CMake has support for specifying which Microsoft C runtime library you want using the variable [`CMAKE_MSVC_RUNTIME_LIBRARY`](https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html).

So maybe we should replace the `CRT_OPT` stuff in `msvc.cmake` with this:

```
if (NOT CMAKE_MSVC_RUNTIME_LIBRARY)
  set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreadedDLL)
endif()
```

By telling CMake which variant we want in its own way hopefully there's no danger of some per-build type default kicking in.